foundations/src/telemetry/memory_profiler.rs

```diff
@@ -182,26 +182,63 @@ fn init_profiler(settings: &MemoryProfilerSettings) -> BootstrapResult<Option<Me
     Ok(Some(MemoryProfiler {
         _seal: Seal,
 
-        #[cfg(feature = "security")]
-        sandbox_profiling_syscalls: settings.sandbox_profiling_syscalls,
+        request_heap_profile: request_sender,
+        _heap_profiling_thread_handle: Arc::new(Mutex::new(heap_profile_thread_handle)),
     }))
 }
 
-fn collect_heap_profile() -> Result<String> {
+#[cfg(feature = "security")]
+fn receive_profiling_thread_setup_msg(
+    setup_complete_receiver: mpsc::Receiver<anyhow::Result<()>>,
+) -> anyhow::Result<()> {
+    match setup_complete_receiver.recv() {
+        Ok(Ok(())) => {}
+        Ok(Err(setup_err)) => {
+            return Err(setup_err);
+        }
+        Err(std::sync::mpsc::RecvError) => {
+            bail!("Profiling thread disconnected before finishing setup")
```

Comment:
will this show the actual error?

Reply:
As far as I can tell, the Error returned by this method will bubble all the way up the call stack to where the telemetry server is initialized